catalog/transformer/catalog-transformer-nitf/src/main/java/ddf/catalog/transformer/nitf/NitfInputTransformer.java

```diff
@@ -0,0 +1,608 @@
+
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.GeometryFactory;
+import com.vividsolutions.jts.geom.LinearRing;
+import com.vividsolutions.jts.geom.MultiPolygon;
+import com.vividsolutions.jts.geom.Polygon;
+import com.vividsolutions.jts.geom.PrecisionModel;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.dynamic.api.DynamicMetacard;
+import ddf.catalog.data.dynamic.api.MetacardFactory;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transform.InputTransformer;
+
+/**
+ * Converts NITF images into a Metacard.
+ */
+public class NitfInputTransformer implements InputTransformer {
+
+    private static final String ID = "ddf/catalog/transformer/nitf";
+
+    private static final String MIME_TYPE = "image/nitf";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NitfInputTransformer.class);
+
+    private static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormat.
+            forPattern("yyyyMMddHHmmss")
+            .withZone(DateTimeZone.UTC);
+
+    private static final GeometryFactory GEOMETRY_FACTORY = new GeometryFactory(new PrecisionModel(
+            PrecisionModel.FLOATING), 4326);
+
+    private static final String NITF_VERSION = "ddf/catalog/transformer/nitf";
+
+    private MetacardFactory metacardFactory;
+
+    /**
+     * Transforms NITF images into a {@link Metacard}
+     */
+    public Metacard transform(InputStream input) throws IOException, CatalogTransformerException {
+        return transform(input, null);
+    }
+
+    public Metacard transform(InputStream input, String id)
+            throws IOException, CatalogTransformerException {
+        if (input == null) {
+            throw new CatalogTransformerException("Cannot transform null input.");
+        }
+
+        final DynamicMetacard metacard = metacardFactory.newInstance(Nitf.NAME);
+        if (metacard == null) {
+            throw new CatalogTransformerException("Unable to create NITF metacard.");
+        }
+
+        List<Polygon> polygonList = new ArrayList<>();
+        // TODO: update to XStream or some xml streaming library to create this metadata
+        StringBuilder metadataXml = new StringBuilder();
+        metadataXml.append("<metadata>\n");
+
+        try {
+            AllDataExtractionParseStrategy parsingStrategy = new AllDataExtractionParseStrategy();
+
+            NitfParserInputFlow inputFlow = new NitfParserInputFlow();
+            inputFlow.inputStream(input)
+                    .imageData()
+                    .fileHeader((header) -> {
+                        setHeaderAttributes(metacard, header);
+                    })
+                    .forEachImage((imagesegmentHeader, imageSegmentData) -> {
+                        handleImageSegmentHeader(metacard, imagesegmentHeader, polygonList);
+                        setMetadataFromImageSegments(metadataXml, imagesegmentHeader);
+                    });
```

Comment:
We don't appear to use get the thumbnail anywhere? Are we going to delegate that off?